SilKit/source/core/vasio/VAsioPeer.cpp

```diff
@@ -109,20 +109,60 @@ auto VAsioPeer::GetSimulationName() const -> const std::string&
 }
 
 void VAsioPeer::SendSilKitMsg(SerializedMessage buffer)
+{
+    auto blob = buffer.ReleaseStorage();
+
+    if (_useAggregation && buffer.GetAggregationKind() == MessageAggregationKind::UserDataMessage)
+    {
+        Aggregate(blob);
+    }
+    else if (_useAggregation && buffer.GetAggregationKind() == MessageAggregationKind::FlushAggregationMessage)
+    {
+        Aggregate(blob); // don't forget to send (current) time sync message
+        Flush();
+    }
+    else
+    {
+        SendSilKitMsgInternal(blob);
+    }
+}
+
+void VAsioPeer::SendSilKitMsgInternal(const std::vector<uint8_t>& blob)
 {
     // Prevent sending when shutting down
     if (!_isShuttingDown && _socket != nullptr)
     {
         std::unique_lock<std::mutex> lock{_sendingQueueMutex};
 
-        _sendingQueue.push_back(buffer.ReleaseStorage());
+        _sendingQueue.push_back(blob);
```

Comment:
This makes a copy because blob is a `const &`. blob should be passed as value and moved. or as an R-value reference.